### PR TITLE
Rollout pods if credentials are changed

### LINKED ISSUE #2180: machine-controller pods should restart on rotation of credentials
### Description of the feature you would like to add / User story
Currently, for few cloud providers, like Openstack and Azure, the credentials for service account expire after certain number of days. It is not possible to get un-expiring passwords for service accounts for some of these cloud providers. So we must rotate the credentials by running `kubeone apply --force-upgrade`

But currently, post rotation, machine-controller pods (controller as well as webhook) do not get restarted and they continue to use old credentials.

So.. we can do one of the below options:

1. Make changes in kubeone logic to restart machine-controller in such cases.
2. Add annotations for [stakater reloader](https://github.com/stakater/Reloader) for on machine-controller and webhook pods so we can add stakater reloader as kubeone addon and it will take care of restarting machine-controller pods when the secrets get updated with new credentials.

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR implements rolling out pods that are using `kubeone-*-credentials` Secrets if credentials are changed.

The implementation is based on applying annotation to all pods that are using `kubeone-*-credentials` Secrets. The annotation contains salt and credentials hashed using the SHA256 algorithm.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2180 

**Does this PR introduce a user-facing change?**:
```release-note
Rollout pods that are using `kubeone-*-credentials` Secrets if credentials are changed
```

### CODE REVIEW

File: pkg/addons/applier.go [hunk 1]
```diff
@@ -341,6 +365,29 @@ func mcCredentialsEnvVars(s *state.State) ([]byte, error) {
...
+	}
+
+	omap := orderedmap.New()
+	for k, v := range creds {
+		omap.Set(k, v)
+	}
+	omap.SortKeys(sort.Strings)
+
+	hash := fmt.Sprintf("kubeone-%s", s.Cluster.CloudProvider.CloudProviderName())
+	for k, v := range creds {
```
Comment (kron4eg) [inline pkg/addons/applier.go:381]: should be iteration over the omap, not creds

File: addons/ccm-digitalocean/ccm-digitalocean.yaml [hunk 2]
```diff
@@ -14,6 +14,8 @@ spec:
     metadata:
       labels:
         app: digitalocean-cloud-controller-manager
+      annotations:
+        "credentials-hash": "{{ .CredentialsCCMHash }}"
```
Comment (ahmedwaleedmalik) [inline addons/ccm-digitalocean/ccm-digitalocean.yaml:18]: Should we add some prefixes to this annotation like "kubeone.k8c.io/"? I think making it explicit that this annotation is managed by KubeOne can be valuable for the end user.

File: pkg/addons/applier.go (hunk 1 above)
Comment (xmudrii) [reply pkg/addons/applier.go]: Fixed.

File: addons/ccm-digitalocean/ccm-digitalocean.yaml (hunk 2 above)
Comment (xmudrii) [reply addons/ccm-digitalocean/ccm-digitalocean.yaml]: I added the `kubeone.k8c.io/` prefix for all KubeOne annotations.

**Pull request merged**